src/StreamingBodyParser/ParserInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace React\Http\StreamingBodyParser;
+
+use Evenement\EventEmitterInterface;
+use React\Http\Request;
+
+interface ParserInterface extends EventEmitterInterface
+{
+    public function __construct(Request $request);
```

Comment:
Another way of doing this is having a public static factory method that takes a request. But that would just wrap passing the request into the constructor. Let me give it a try to see how it works out, I don't mind either way :smile: .